src/lib/ValidateBoard.ts

```diff
@@ -0,0 +1,30 @@
+import Cell from './Cell';
+import SudokuBoard from './Board';
+import { BOARD_SIZE } from './Board';
+
+const validateSection = (section: Cell[]) => {
+  const values = section.map((cell) => cell.value);
+  const uniqueValues = new Set(values);
+  return values.length === uniqueValues.size;
+};
+
+export default (board: SudokuBoard) => {
+  let valid = true;
+
+  for (let i = 0; i < BOARD_SIZE; i++) {
+    const row = board.row(i);
+    const col = board.col(i);
+    const square = board.fetchCellsInSquare(i, i);
+
+    if (
+      !validateSection(row) ||
+      !validateSection(col) ||
+      !validateSection(square)
+    ) {
```

Comment:
```suggestion
    if (
      [row, col, square]
        .filter(item=>!validateSelection(item))
        .length !== 0 // '> 0' also works but this is more optimized
    ) {
```
Copy-pasted `!validateSection(` is awkward & should be replaced with something like the above array.filter(func).length

Reply:
Reasoning behind this is allowing extending the validator with more items to be easier, overall good practice